src/transports/sentry_transport_curl.c

```diff
@@ -247,6 +247,7 @@ sentry__curl_send_task(void *_envelope, void *_state)
         }
     }
 
+    curl_easy_reset(curl);
```

Comment:
Any specific reason you added this?

Reply:
I wanted to see if there was a trivial reason for `ASAN` failing to track `curl` globals when shutting down early (this is only related to the failing `ASAN` test on macOS and unrelated to the changes here, but it will fail again, so probably have to fix it here).